tensorboard/webapp/notification_center/_redux/testing.ts

```diff
@@ -33,3 +42,38 @@ export function buildStateFromNotificationState(
 ): State {
   return {[NOTIFICATION_FEATURE_KEY]: runsState};
 }
+
+@Injectable()
+export class TestingNotificationCenterDataSource
+  implements NotificationCenterDataSource {
+  fetchNotifications() {
+    return of({
+      notifications: [],
+      error: '',
+    });
+  }
+}
+
+export function provideTestingNotificationCenterDataSource() {
+  return [
+    TestingNotificationCenterDataSource,
+    {
+      provide: NotificationCenterDataSource,
+      useExisting: TestingNotificationCenterDataSource,
+    },
+  ];
+}
+
```

Comment:
these should go into the `_data_source/testing.ts`.

Reply:
Why is that? Given that this is used in effects.ts?
Moved in _data_source/ results in re-exporting in index.ts?

Reply:
Sure but DataSource should provide the testing utility to mocking itself out so when you change the API contract, you know where and how to do it without going to every consumer of the DataSources.

Reply:
Gotcha. I would like to leave this part (moving the datasource testing) to https://github.com/japie1235813/tensorboard/pull/12 cause it's about testing. Otherwise too work duplication (BUILD, adding new file etc) and much more conflict.